frontend/src/components/TripCard.tsx

```diff
@@ -0,0 +1,156 @@
+import {
+  View,
+  Text,
+  Image,
+  StyleSheet,
+  TouchableHighlight,
+} from "react-native";
+import { useState, useEffect } from "react";
+import { Float } from "react-native/Libraries/Types/CodegenTypes";
+import { Trip } from "../types";
+import { Octicons, MaterialCommunityIcons } from '@expo/vector-icons';
+import { Link, router, Stack } from "expo-router";
+
+interface tripProps {
+  trip: Trip;
+  height?: number; // Optional height prop
+  width?: number; // Optional width prop
+}
+
+export const TripCard: React.FC<tripProps> = ({ 
+  trip,
+  height = 300,
+  width = 350,
+ }) => {
+  const [tripImage, setTripImage] = useState(trip.image);
+  const [tripLocation, setTripLocation] = useState(trip.location);
+  const [tripName, setTripName] = useState(trip.name);
+  const [tripStartDate, setTripStartDate] = useState(trip.startDate);
+  const [tripEndDate, setTripEndDate] = useState(trip.endDate);
+  const [fontTripName, setFontTripName] = useState(Math.max(18, (height*18)/300));
+  const [fontTripDetail, setFontTripDetail] = useState(Math.max(13, (height*14)/300));
+  const [location, setLocation] = useState(trip.location.citystate);
+  const noImage =
+    "https://st4.depositphotos.com/14953852/24787/v/450/depositphotos_247872612-stock-illustration-no-image-available-icon-vector.jpg";
+
+  const defaultAvatar = 
+    "https://static1.colliderimages.com/wordpress/wp-content/uploads/2022/02/avatar-the-last-airbender-7-essential-episodes.jpg";
+
+  const onPressTripCard = () => {
+    // console.log("You pressed this card");
+    const route = `/trips/${trip.id}`;
+    router.push(route);
+  };
+
+  // Format the Date of Trip Card
+  const tripDate = (date: Date) => {
+    if (!date || !(date instanceof Date) || isNaN(date.getTime())) {
+      return "Invalid date"; // Handle invalid dates
+    }
+    const month = date.toLocaleString("default", { month: "short" });
+    return `${date.getDate()} ${month}, ${date.getFullYear()}`;
+  };
+
+  // Calculate image height as 2/3 of the card's height
+  const imageHeight = (height * 2) / 3;
+  const textSize = height / 18;
+  const tripState = tripLocation.citystate.split(", ");
+
+  useEffect(() => {
+    //fontTripDetail == height of the text --> 0.75*fontTripDetail ~ width of the text
+    if(tripLocation.citystate.length*fontTripDetail*0.75 >= width)
+    if(tripState.length >= 2)
+      setLocation(tripState[tripState.length - 2] + ", " + tripState[tripState.length - 1]);
```

Comment:
<img width="326" alt="Screenshot 2024-05-02 at 2 00 46 PM" src="https://github.com/trihoang0809/tourific/assets/87792873/fcff7e6e-4b22-4a72-b14e-c2e681c3b0ed">
I don't think it is working as expected as this card width increases with the location field

Reply:
You can simply combine numberOfLines and style `width: 100` in the Text field that contains location

Reply:
> <img alt="Screenshot 2024-05-02 at 2 00 46 PM" width="326" src="https://private-user-images.githubusercontent.com/87792873/327586352-fcff7e6e-4b22-4a72-b14e-c2e681c3b0ed.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ2ODU5MTYsIm5iZiI6MTcxNDY4NTYxNiwicGF0aCI6Ii84Nzc5Mjg3My8zMjc1ODYzNTItZmNmZjdlNmUtNGIyMi00YTcyLWIxNGUtYzJlNjgxYzNiMGVkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTAyVDIxMzMzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk5ZGJkNDFmMTQ2YThkM2Y0M2JjMjBjYzM3YWYwMGFlZjI3Zjc3OTQ5ODI3NWY3Y2E5NjRjMzI1Y2FlODYzMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WN_zlDjV95V5tf736LmByY_ukpbnaI8S8tVYjjY61fs"> I don't think it is working as expected as this card width increases with the location field

It works well there since I just restrict it to not exceeding the width of the card.



> You can simply combine numberOfLines and style `width: 100` in the Text field that contains location

I'll change to this